routes/helpers/getMovieDetails.js

```diff
@@ -0,0 +1,49 @@
+/* dependencies */
+const async = require('async');
+const keystone = require('keystone');
+
+/* models */
+const Director = keystone.list('Director');
+const Theme = keystone.list('Theme');
+const Genre = keystone.list('Genre');
+
+let movieDetails;
+
+const getDirector = (cb) => {
+
+  Director.model.findOne({ _id: movieDetails.director, }).exec((err, director) => {
+    if (err) return cb(err);
+    return cb(null, director);
+  });
+};
+
+const getThemes = (cb) => {
+
+  Theme.model.find({ _id: { $in: movieDetails.themes, }, }).exec((err, themes) => {
+    if (err) return cb(err);
+    return cb(null, themes);
+  });
+};
+
+const getGenres = (cb) => {
+
+  Genre.model.find({ _id: { $in: movieDetails.genre, }, }).exec((err, genres) => {
+    if (err) return cb(err);
+    return cb(null, genres);
+  });
+};
+
+module.exports = (movie, cb) => {
+
+  movieDetails = movie;
+
+  async.parallel([ getDirector, getThemes, getGenres, ], (err, result) => {
+
+    if (err) return cb(err);
+    return cb(null, {
+      director: result[0],
```

Comment:
Hi,

Here is the top level: http://mongoosejs.com/docs/api.html#query_Query-populate, and the more detailed api: http://mongoosejs.com/docs/api.html#query_Query-populate

I am assuming its the monogoose api that you are using..

Finally, you want to be using `findById` http://mongoosejs.com/docs/api.html#model_Model.findById